### PR TITLE
fix: multiple fix in `Iatp` modules

### PR DESCRIPTION
**Pull request by bscholtes1A**
## What this PR changes/adds

 - Avoid `NullPointerException` in `TokenValidationRulesRegistryImpl` if there is no rules registered for a given context
 - In `JwtPresentationVerifier`:
 - do not validate VC `aud` claim, as they don't have one
 - do not validate VP `sub` claim, as they don't have one
 - **MAKE VERIFIER CONTEXT IMMUTABLE**: this was causing very dubious behavior as `JwtPresentationVerifier.verify` was modifying the `audience` in the `VerificationContext` in the 
recursive call but it was never restored to the initial and expected value for the subsequent presentation verifications! So basically, first presentation verification was fine, but all consecutives were failing because the context audience was changed at the end of the first execution!
 - In `DidPublicKeyResolverImpl`: improve error logging when the provided did is null (happens when the validated token does not have the `kid` claim)
 - Make `JwtToVerifiableCredentialTransformer` and `JwtToVerifiablePresentationTransformer` more robust by also handling the cases where `credentialSubject` and `credentialStatus` are list!
 - Fix missing `credentialSubject.id` mapping in `JwtToVerifiableCredentialTransformer`
 - Fix the serialization of `VerifiableCredential` `type` field: current implem was giving `types` instead of `type` (see spec: https://www.w3.org/TR/vc-data-model-2.0/#example-a-simple-example-of-a-verifiable-presentation)
 
## Why it does that

Fix.

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IdentityAndTrustExtension.java [hunk 1]
```diff
@@ -125,12 +124,7 @@ public void initialize(ServiceExtensionContext context) {
         rulesRegistry.addRule(IATP_SELF_ISSUED_TOKEN_CONTEXT, new AccessTokenNotNullRule());
 
         // add all rules for validating VerifiableCredential JWTs
-        rulesRegistry.addRule("iatp-vc", (toVerify, additional) -> Optional.ofNullable(toVerify.getStringClaim(JWTClaimNames.SUBJECT)).map(s ->
```
Comment (bscholtes1A) [inline extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IdentityAndTrustExtension.java:128]: There is not `sub` in VC

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiableCredential.java [hunk 2]
```diff
@@ -41,6 +42,8 @@ public class VerifiableCredential {
     public static final String VERIFIABLE_CREDENTIAL_PROOF_PROPERTY = "https://w3id.org/security#proof";
     private List<CredentialSubject> credentialSubject = new ArrayList<>();
     private String id; // must be URI, but URI is less efficient at runtime
+
+    @JsonProperty("type")
```
Comment (paullatzelsperger) [inline spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiableCredential.java:46]: nit: should we do the same for `credentialSubject`? i.e. call the field `credentialSubjects` (plural), but annotating with a JsonProperty to conform to the spec

**Review by paullatzelsperger — APPROVED** [review]
:rocket:

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiableCredential.java (hunk 2 above)
Comment (bscholtes1A) [reply spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiableCredential.java:48]: done

**Pull request merged**